rca/static_src/javascript/programmes/components/ProgrammesResults/ProgrammesResults.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { programmePageShape } from '../../programmes.types';
+
+import Icon from '../Icon/Icon';
+import ProgrammeTeaser from './ProgrammeTeaser';
+
+const getResultsCount = (count) => {
+    switch (count) {
+        case 0: {
+            return 'No results match your search';
+        }
+        case 1: {
+            return '1 result matches your search';
+        }
+        default: {
+            return `${count} results match your search`;
+        }
+    }
+};
+
+/**
+ * A list of programmes matching a search or filter.
+ * The list auto-magically appears when matches are found.
+ */
+const ProgrammesResults = ({ programmes, hasActiveSearch }) => {
+    const [activeProgramme, setActiveProgramme] = useState(null);
+    if (!hasActiveSearch) {
+        return null;
+    }
+
+    const count = getResultsCount(programmes.length);
+    return (
+        <>
+            <div className="programmes-results bg bg--dark section">
+                <div className="grid">
+                    <div className="programmes-results__actions">
+                        <button
+                            type="button"
+                            className="button programmes-results__back body body--one"
+                        >
+                            <Icon
+                                name="arrow"
+                                className="programmes-results__back__icon"
+                            />
+                            <span className="programmes-results__back__text">
+                                Back
+                            </span>
+                        </button>
+                    </div>
+                    <p
+                        className="heading heading--five programmes-results__count"
+                        role="alert"
+                    >
+                        {count}
+                    </p>
+                </div>
+                {programmes.length === 0 ? null : (
+                    <div className="grid">
+                        <div className="programmes-results__list">
+                            {programmes.map((prog) => {
+                                const onHocus = () =>
```

Comment:
At first i thought this was a typo but i'm now thinking you've created a hybrid of hover and focus?

Reply:
Hahaha yes, this is a name I’ve seen when you want to apply some behavior or styles both to focus and hover, usually as a mixin:

```scss
@mixin hocus() {\
    &:focus,
    &:hover {
        @contents;
    }
}

a {
    @include hocus() {
        cursor: pointer;
    }
}
```

I’ll rename this, it’s already hard enough to follow.